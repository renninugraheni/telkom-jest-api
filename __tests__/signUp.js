const ep = require('../endpoints/epUsers');
const data = require('../data/data');
const schema = require('../schema/schemaPosts');

const { matchers } = require('jest-json-schema');

expect.extend(matchers);

describe('test api sign up user', () => {
    test('user can successfully sign up using valid data', async () => {
        const response = await ep.epPost(data.dataPost);
        expect(response.status).toEqual(200);
        expect(response.body).toMatchSchema(schema.schema);
    });
});

describe('negative test api sign up user', () => {
    test('user failed sign up using empty password', async () => {
        const response = await ep.epPost(data.emptyData);
        expect(response.status).toEqual(422);
        expect(response.body).toMatchSnapshot();
    });
});