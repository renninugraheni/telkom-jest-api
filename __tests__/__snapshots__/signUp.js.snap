// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`negative test api sign up user user failed sign up using empty password 1`] = `
Object {
  "errors": Object {
    "first_name": Array [
      "can't be blank",
    ],
    "password": Array [
      "Can't be blank",
    ],
  },
  "message": "Sign up failure",
}
`;
